### PR TITLE
add support for event-traits type

### PR DESCRIPTION
**Pull request by dlasky**
Adds support to the IR for event-traits and corresponding generation/parsing

### CODE REVIEW

File: packages/destination-subscriptions/src/types.ts [hunk 1]
```diff
@@ -65,6 +73,7 @@ export type ConditionType =
 	| 'event-type'
 	| 'event'
 	| 'event-property'
+	| 'event-trait'
 	| 'event-context'
 
 export type PropertyConditionType = 'event-property' | 'event-context'
```
Comment (mshwery) [inline packages/destination-subscriptions/src/types.ts:79]: Do we need to add `event-trait` here?

**Review by mshwery — APPROVED** [review]
lgtm, do we need to update tests?

File: packages/destination-subscriptions/src/types.ts (hunk 1 above)
Comment (dlasky) [reply packages/destination-subscriptions/src/types.ts:79]: when we parse this back out in app there needs to be some way to differentiate at the UI level, currently thats done via this type explicitly

**Comment by dlasky** [comment]
yep just wanted to get 👀 before proceeding

**Comment by dlasky** [comment]
@mshwery one more look if you would, notably: d8eb2b2

**Review by mshwery — APPROVED** [review]

**Review by lcampos — APPROVED** [review]

**Pull request merged**